src/main/java/study/step3/CarRacing.java

```diff
@@ -0,0 +1,52 @@
+package study.step3;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Scanner;
+
+public class CarRacing {
+
+    private static final String NULL_INPUT = "입력값이 null 입니다";
+    private static final String CAR_QUESTION = "자동차 대수는 몇 대 인가요?";
+    private static final String ROUND_QUESTION = "시도할 회수는 몇 회 인가요?";
+    private static final Scanner scanner = new Scanner(System.in);
+
+    public static void main(String[] args) {
+        int carNumber = getNumber(CAR_QUESTION);
+        int roundNumber = getNumber(ROUND_QUESTION);
+        printRaceRecords(raceManager().manageRace(carNumber, roundNumber));
+    }
+
+    private static void printRaceRecords(String raceRecords) {
+        System.out.println("실행 결과\n" + raceRecords);
+    }
+
+    private static int getNumber(String question) {
+        System.out.println(question);
+        return validation(scanner.nextLine());
+    }
+
+    private static int validation(String number) {
+        return requireNumber(requireNonNull(number));
+    }
+
+    private static String requireNonNull(String number) {
+        if (Objects.isNull(number)) {
+            throw new IllegalArgumentException(NULL_INPUT);
+        }
+        return number;
+    }
+
+    private static int requireNumber(String number) {
+        return Optional.of(Integer.parseInt(number))
+                .orElseThrow(NumberFormatException::new);
```

Comment:
파싱할 수 없는 것과 값 자체가 없는 것은 다르다고 생각하는데요
이러한 부분을 나눠서 `Optional`을 사용한다면 아래와 같은 코드가 될 것 같아요
```suggestion
        return Optional.of(number)
                .map(Integer::parseInt)
                .orElseThrow(() -> new IllegalArgumentException("require number : " + number));
```